### DISCUSSION TITLE
Shutter stopping in the middle forget Calibration

### DISCUSSION THREAD

**Discussion by RepiZoli** [arendst/Tasmota#22079] [Q&A] [answered]
Hi all 

I made a Calibration for my shutters as Follows:https://tasmota.github.io/docs/Blinds-and-Shutters/#calibration

 ShutterSetClose
ShutterOpenDuration 21  " its the opening time
ShutterOpen
ShutterSetOpen 
ShutterCloseDuration 21

It works perfectly, but after some day (maybe a blackout)  you open with the switch, it opens only still half of the shutter.. And if you check in tasmota, it shows the shutter is fully open.
 After I make the calibration, it work again perfectly
Where can I made a mistake?
 Can you help me?

Thanks in advance

Repizoli

**Comment by valdus2** [comment, marked as answer]
This has happened to me too using some Sonoff's Dual R3 v2. They just lost their calibration.
But to me happened in this time frame  [#21966](https://github.com/arendst/Tasmota/pull/21966)
Now I have updated them and I wait...

**Reply by RepiZoli** [reply]
Hi have a same Sonoff Dual, but I do not understand fully your answer
1. What was the update exactly?
2. It Is working normal now?
Thanks in advance

**Reply by valdus2** [reply]
It happened to me after 50 days online....They were in v13.3.0.
Now I have updated them to the latest development version,
and I wait to see their behavior...

**Comment by stefanbode** [comment]
Can you check the uptime of the device. The only situation I can think about that the known and the real position gets this far disconnected is a crash during operation with a reboot.
The position if the shutter is stored into the flash AFTER the movement is complete. This is no issue in most cases. But for sure you don't want to have a crash during operation or a power loss.

**Reply by joba-1** [reply]
only used this in early experiments with ESPs a long time ago, but wouldn't [noinit-dram](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/memory-types.html#noinit-dram) or RTC_NOINIT_ATTR be a solution for at least the software resets / crashes?

**Reply by stefanbode** [reply]
There is some rtc Memory that would also survive. Anyhow it does not help because this only correct if you assume the shutter immediately stops on a crash. This is not always the matter. Better approach is to hunt every crash and remove it.

**Comment by valdus2** [comment]
I have forgotten to say, that for me the problem was resolved after updating them. 
70 days online and working OK. No calibration has been lost.

![Untitled](https://github.com/user-attachments/assets/b4619525-ab99-4929-bfbf-d62eeb392032)

**Reply by RepiZoli** [reply]
Hi Thanks valus2

So you updated to v14.2.0.6 The actual release is : Release binaries for Tasmota firmware 14.4.1
 I will use this.

Thanks Zoli

**Comment by RepiZoli** [comment]
Hi All 
 At last I begun to update, however the wonder if my parameters remain the same like "Interlock ON" See more: https://tasmota.github.io/docs/Blinds-and-Shutters/#shutter-modes

It seems , but I am not sure. How can I check it? 

Thanks Zoli

**Reply by stefanbode** [reply]
All parameters are the same. If we make changes then there is an automatic upgrade in TASMOTA. Just enjoy and have fun

**Reply by RepiZoli** [reply]
Thanks a lot

**Comment by RepiZoli** [comment]
Hi I can update all but 1 of my shutter  with OTA most of them tasmota 12 --> tasmota 14

This one has an Tasmota 9.3 It gives the following failure:

Sonoff Dual R3 Module
ccccc
Upload Failed

**Upload buffer miscompare**

Main Menu
[Tasmota 9.3.1.2 by Theo Arends](https://bit.ly/tasmota)

Any Idea
 Thanks in advance

**Reply by sfromis** [reply]
Make sure to use the correct file for OTA upgrade, always the URL ending with `.gz` (preferably not downloading first, then you'd need an extra minimal step yourself instead of automatic).

**Reply by RepiZoli** [reply]
Hi  I tried it with 
The OTA: <html>
<body>

http://ota.tasmota.com/tasmota32/release/tasmota32.bin
--



</body>
</html>

I also download the following: <html>
<body>

http://ota.tasmota.com/tasmota/release/tasmota.bin.gz
--



</body>
</html>

But I got the same failure.. Any idea.

 Thanks in advance

Zoli

**Reply by sfromis** [reply]
See https://github.com/arendst/Tasmota/discussions/18983
